lib/matplotlib/axes/_axes.py

```diff
@@ -7369,12 +7396,44 @@ def _kde_method(X, coords):
             kde = mlab.GaussianKDE(X, bw_method)
             return kde.evaluate(coords)
 
-        vpstats = cbook.violin_stats(dataset, _kde_method, points=points)
-        return self.violin(vpstats, positions=positions, vert=vert,
+        def _resolve_duplicate_alias(func_obj_list):
+            unique_alias_set = set()
+            for func_obj in func_obj_list:
+                while func_obj.alias in unique_alias_set:
+                    func_obj.alias += 'x'
+                unique_alias_set.add(func_obj.alias)
```

Comment:
No. unique_alias_set keeps track of all alias so far. The actual alias (modified so that they are unique) is in the ViolinStatFunc object itself. I think this defensive programming mechanism helps safe-guarding against user inputs where multiple input functions have repeating aliases. These aliases later on become the keys in the dict returned by violin function. Thoughts?